dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityLiveMode.java

```diff
@@ -0,0 +1,193 @@
+package com.dotcms.rendering.velocity.servlet;
+
+import com.dotcms.enterprise.LicenseUtil;
+import com.dotcms.rendering.velocity.util.VelocityUtil;
+import com.dotcms.visitor.domain.Visitor;
+
+import com.dotmarketing.beans.Host;
+import com.dotmarketing.beans.Identifier;
+import com.dotmarketing.beans.UserProxy;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.BlockPageCache;
+import com.dotmarketing.business.BlockPageCache.PageCacheParameters;
+import com.dotmarketing.business.CacheLocator;
+import com.dotmarketing.business.web.WebAPILocator;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.factories.ClickstreamFactory;
+import com.dotmarketing.filters.CMSUrlUtil;
+import com.dotmarketing.portlets.htmlpageasset.model.IHTMLPage;
+import com.dotmarketing.portlets.rules.business.RulesEngine;
+import com.dotmarketing.portlets.rules.model.Rule;
+import com.dotmarketing.util.Config;
+import com.dotmarketing.util.Logger;
+import com.dotmarketing.util.PageMode;
+import com.dotmarketing.util.WebKeys;
+
+import java.io.IOException;
+import java.io.StringWriter;
+import java.io.Writer;
+import java.util.Optional;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+
+import org.apache.velocity.context.Context;
+
+import com.liferay.portal.model.User;
+
+public class VelocityLiveMode extends VelocityModeHandler {
+
+
+
+    private final HttpServletRequest request;
+    private final HttpServletResponse response;
+    private final PageMode mode = PageMode.LIVE;
+    private final String uri;
+    private final Host host;
+
+
+    
+    
+    public VelocityLiveMode(HttpServletRequest request, HttpServletResponse response, String uri, Host host) {
+        this.request = request;
+        this.response = response;
+        this.uri = uri;
+        this.host = host;
+    }
+
+    public VelocityLiveMode(HttpServletRequest request, HttpServletResponse response) {
+        this(request, response, request.getRequestURI(), hostWebAPI.getCurrentHostNoThrow(request));
+    }
+
+
+    @Override
+    public final void serve() throws Exception {
+
+        serve(response.getWriter());
+
+    }
+    @Override
+    public final void serve(final Writer out) throws DotDataException, IOException, DotSecurityException {
+
+        LicenseUtil.startLiveMode();
+        try {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Change this "try" to a try-with-resources. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2093)